File: Cpp/Backpack.cpp
class Solution {
public:
    /**
     * @param m: An integer m denotes the size of a backpack
     * @param A: Given n items with size A[i]
     * @return: The maximum size
     */
    int backPack(int m, vector<int> A) {
        // write your code here
        int n = A.size();
        vector<int> vec(m+1);
        for (int i = 0; i < n; i ++) {
            for (int j = m; j >= 0; j --) {
                if (j >= A[i]) vec[j] = max(vec[j-A[i]] + A[i], vec[j]);
            }
        }
        return vec[m];
    }
};

File: Java/Backpack.java
public class Solution {
    /**
     * @param m: An integer m denotes the size of a backpack
     * @param A: Given n items with size A[i]
     * @return: The maximum size
     */
    public int backPack(int m, int[] A) {
        // write your code here
        int n = A.length;
        boolean[][] dp = new boolean[n + 1][m + 1];
        dp[0][0] = true;
        for (int i = 0; i < n; i ++) {
            for (int j = 0; j <= m; j ++) {
                if (dp[i][j] == true && j + A[i] <= m) {
                    dp[i + 1][j + A[i]] = true;
                }
                if (dp[i][j]) dp[i + 1][j] = true;
            }
        }
        for (int i = m; i >= 0; i --) {
            if (dp[n][i]) return i;
        }
        return 0;
    }
}
